python/lsst/daf/butler/tests/_examplePythonTypes.py

```diff
@@ -186,30 +193,37 @@ class MetricsExample:
         Arbitrary array data.
     """
 
-    def __init__(self, summary=None, output=None, data=None):
+    def __init__(
+        self,
+        summary: dict[str, Any] | None = None,
+        output: dict[str, Any] | None = None,
+        data: list[Any] | None = None,
+    ) -> None:
         self.summary = summary
         self.output = output
         self.data = data
 
-    def __eq__(self, other):
+    def __eq__(self, other: Any) -> bool:
+        if not (hasattr(other, "summary") and hasattr(other, "output") and hasattr(other, "data")):
+            return NotImplemented
```

Comment:
May be more efficient to catch `AttributeError` than do explicit check, though I'm sure it does not matter here.

Reply:
My recollection is that mypy doesn't work with try/catch but does work with hasattr.

Reply:
Maybe mypy  0.991 handles `__eq__` differently because now mypy doesn't complain at all about `other: Any` and lets everything through. I've used try/except now.